src/withLocation.js

```diff
@@ -0,0 +1,57 @@
+import omitBy from 'lodash/omitBy';
+import isNil from 'lodash/isNil';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { parse, stringify } from 'query-string';
+import { compose } from 'redux';
+import { withRouter } from 'react-router';
+
+function withLocation(WrappedComponent) {
```

Comment:
@mkuklis, about your concern about the name. I was thinking about `withHistory`. But eventually, think that `withLocation` might be the more appropriate here as what it really provides is the way to change location.